Add z3-solver as dependency to dynamo tests

ezyang opened the pull request:
Test Plan: sandcastle

Reviewed By: malfet, huydhn

Differential Revision: D47438456



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305 @ipiszy

aaronenyeshi reviewed (approved):
already reviewed internally

ezyang committed:
Add z3-solver as dependency to dynamo tests (#105640)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/105640

Test Plan: sandcastle

Reviewed By: malfet, albanD, huydhn, atalman

Differential Revision: D47438456

fbshipit-source-id: 3b3eee3a7cafee09b83f1ee11bd4bbd3a8c68e0d
